Delete redirected files - update redirection.json

tompratt-AQ opened the pull request:
# Title
Delete redirected files - update redirection.json

## Summary
Did the following to cleanup redirected files:

1. Deleted all files that had only a redirect_url (pages that redirected with no content of their own).
2. Made an entry in redirection.json for each of the files.
3. Confirmed that all new redirection links are valid.
4. Made sure that all entries in redirection.json are alphabetized based on the source URL.

Fixes #1773 

## Suggested Reviewers
@mairaw

tompratt-AQ committed:
Delete redirected files - update redirection.json

dnfclas commented:
@tompratt-AQ,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by .NET Foundation. We will now review your pull request.
_Thanks,
.NET Foundation Pull Request Bot_

mairaw commented on .openpublishing.redirection.json:
```diff
@@ -195,18 +189,193 @@
+        "redirect_url": "/dotnet/articles/core/windows-prerequisites"
+    },
+    {
+        "source_path": "docs/core/testing/using-mstest-on-windows.md", 
```
this one also had the following metadata redirect_document_id: TRUE, so "redirect_document_id": true should also be added here.

tompratt-AQ replied:
Great catch! I fixed this in the second commit -  "Added redirect_document_id=true per review comment".

tompratt-AQ commented:
@mairaw  Here are a few questions/comments to consider during the review:

1. The DOCS/About folder is now empty.  Should it be deleted?
2. The file [asynchronous-programming-with-async-and-await](https://docs.microsoft.com/en-us/dotnet/articles/csharp/programming-guide/concepts/async/asynchronous-programming-with-async-and-await) redirects TO the [main csharp index ](https://docs.microsoft.com/en-us/dotnet/articles/csharp/async/), wouldn't make more sense for it to link TO the Programming [Guide Concepts index](https://docs.microsoft.com/en-us/dotnet/articles/csharp/programming-guide/concepts/async/index) that has the same information?
3. I deleted the TOC file in the DOCS/csharp/programming-guide/linq-query-expressions folder, because all of the files it references are being deleted.
4. This file: docs/visual-basic/developing-apps/using-ide/refactoring-and-rename-dialog-box.md
Has a full page of content but has a redirect URL in it's header. Should  a redirect be entered in the JSON file and the page be deleted?
6. What about this file: https://docs.microsoft.com/en-us/dotnet/articles/core/tutorials/aspnet-core
Should it be deleted - it's basically just a redirect:

mairaw commented:
@tompratt-AQ 

1. It should be automatically deleted by GitHub when it notices all files are gone
2. @BillWagner can you comment on this one? I think we chose to link to the newer articles vs. the migrated content
3. Sounds good
4. @GuardRex already took care of those with #1774.  
5. Keep that one for now.

tompratt-AQ committed:
Added redirect_document_id=true per review comment

mairaw committed:
Merge branch 'master' into Issue-1773-Remove-redirected-files

tompratt-AQ committed:
Merge https://github.com/tompratt-AQ/docs into Issue-1773-Remove-redirected-files

tompratt-AQ committed:
Merge branch 'Issue-1773-Remove-redirected-files' of https://github.com/tompratt-AQ/docs into Issue-1773-Remove-redirected-files

tompratt-AQ committed:
Fixed merge conflict in redirect file.

tompratt-AQ commented:
@mairaw  I fixed the merge conflict in the redirect.json file, so this should be ready to go.

mairaw commented on docs/csharp/programming-guide/linq-query-expressions/toc.md:
```diff
@@ -1,19 +0,0 @@
-# [LINQ Query Expressions](index.md)
```
you're now getting a warning on the build report because this file was removed. I think you need to remove the reference to this folder from https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/toc.md

tompratt-AQ replied:
@mairaw Thanks for identifying the error. I removed the entry from the TOC, and made sure that there weren't others that were in my list of deleted files for this PR.

tompratt-AQ replied:
@mairaw Not ready yet! I see that I didn't edit the TOC correctly. My apologies, it's the first TOC I've edited and this was the first file in the TOC.

tompratt-AQ committed:
Removed deleted file from TOC.

tompratt-AQ committed:
Deleted duplicate code snippets and fixed TOCs.

tompratt-AQ commented:
@mairaw I fixed the TOC as you suggested and deleted the rest of the files in the csharp\programming-guide\linq-query-expressions directory, the files were not linked to. They were all duplicates of code snippets that are in samples\snippets\csharp\concepts\linq which are being used.

mairaw reviewed (approved):
That's great!

mairaw commented on docs/csharp/programming-guide/concepts/linq/toc.md:
```diff
@@ -1,4 +1,4 @@
-# [Language-Integrated Query (LINQ)](index.md)
+# [LINQ Query Expressions](index.md) 
```
just noticed something. this file shouldn't be changed.

tompratt-AQ committed:
Corrected link text per review comment.

tompratt-AQ commented:
@mairaw Amazing catch! I forgot to undo all changes to that file by changing the link text back. (fingers crossed) I think this one is finally ready.

tompratt-AQ commented:
I'm closing this Pull Request. Due to it's age, the merge conflicts where too difficult to resolve. I am submitting a new PR with all of the same changes to fix issue 1773.
